lib/mix/lib/mix/compilers/elixir.ex

```diff
@@ -45,27 +45,31 @@ defmodule Mix.Compilers.Elixir do
       |> MapSet.difference(all_paths)
       |> MapSet.to_list()
 
-    changed =
+    {changed, sources_stats} =
       if force do
         # A config, path dependency or manifest has
         # changed, let's just compile everything
-        MapSet.to_list(all_paths)
+        all_paths = MapSet.to_list(all_paths)
+        {all_paths, mtimes_and_sizes(all_paths)}
```

Comment:
Let's not call mtimes_and_sizes here. Instead let's do a simple comprehension:

    for path <- all_paths, into: %{}, do: {path, Mix.Utils.last_modified_and_size(path)}

We don't need to worry about dupes here and this way we don't need to change mtimes_and_sizes.